tests/test_population.py

```diff
@@ -78,122 +59,121 @@ def raise_function(_):
-        pop2 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
-        pop3 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
-        assert len(pop1) == 200
-        assert len(pop2.survive(fraction=0.5)) == 100
-        assert len(pop3.survive(fraction=0.1, luck=True)) == 20
-
-    def test_survive_n_and_p_works(self):
-        pop1 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
-        pop2 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
-        pop3 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
+        assert len(pop3.survive(n=75, luck=True)) == 75
+
+    def test_survive_p_works(self, simple_chromosomes, simple_evaluation_function):
+        pop1 = Population(chromosomes=simple_chromosomes, eval_function=simple_evaluation_function)
+        pop2 = Population(chromosomes=simple_chromosomes, eval_function=simple_evaluation_function)
+        pop3 = Population(chromosomes=simple_chromosomes, eval_function=simple_evaluation_function)
+        assert len(pop1) == len(simple_chromosomes)
+        assert len(pop2.survive(fraction=0.5)) == len(simple_chromosomes) * 0.5
+        assert len(pop3.survive(fraction=0.1, luck=True)) == len(simple_chromosomes) * 0.1
+
+    def test_survive_n_and_p_works(self, simple_evaluation_function):
+        chromosomes = list(range(200))
+        pop1 = Population(chromosomes=chromosomes, eval_function=simple_evaluation_function)
+        pop2 = Population(chromosomes=chromosomes, eval_function=simple_evaluation_function)
+        pop3 = Population(chromosomes=chromosomes, eval_function=simple_evaluation_function)
         assert len(pop1.survive(fraction=0.5, n=200)) == 100
         assert len(pop2.survive(fraction=0.9, n=10)) == 10
         assert len(pop3.survive(fraction=0.5, n=190, luck=True)) == 100
 
-    def test_survive_throws_correct_errors(self):
+    def test_survive_throws_correct_errors(self, simple_population):
         """If the resulting population is zero or larger than initial we need to see errors."""
-        pop1 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
         with raises(RuntimeError):
-            pop1.survive(n=0)
-        pop2 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
+            simple_population.survive(n=0)
         with raises(ValueError):
-            pop2.survive(n=250)
-        pop3 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
+            simple_population.survive(n=250)
         with raises(ValueError):
-            pop3.survive()
+            simple_population.survive()
 
 
-class TestPopulationBreed(TestPopulation):
+class TestPopulationBreed:
 
-    def test_breed_amount_works(self):
-        pop1 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
+    def test_breed_amount_works(self, simple_chromosomes, simple_evaluation_function):
```

Comment:
it feels a bit magic-y if you've never used pytest like this before, but it seems to work alright.